### DISCUSSION TITLE
Hidden Model Properties Erase Input

### DISCUSSION THREAD

**Discussion by Coyote6** [livewire/livewire#3020] [7. Bugs (archived)]
### Description
When hidden properties are set on the model, such as below:
```php
class User extends Authenticatable {
	
    protected $hidden = [
        'password',
        'remember_token',
    ];
    
}
```

 And a property on the Livewire property uses the model for type casting:
```php
class Test extends Component {
	public User $creating;
}
```
 Then any attempts to use the name of the hidden field gets erased.  Might just be a limitation of the use of the model. 
 
### Exact steps to reproduce
1. Create a model with a hidden field (or use an existing one).
2. Create a component with a model type cast on a property.
3. Try to use a wire:model of the hidden property name.

### Stripped-down, copy-pastable code snippets
Livewire Component
```php
namespace App\Http\Livewire;

use App\Models\User;
use Livewire\Component;

class Test extends Component {
	
	public User $creating;
	
	public function rules () {
		return [
			'creating.password' => 'required'
		];
	}
	
	public function mount() {
		$this->creating = User::make();
	}

	public function render() {
		return view('livewire.test');	
	}
	
}
```
Blade Template
```html
<div>
	<input wire:model="creating.password" name="password" type="password">
</div>
```

A workaround is to rename the property, but would prefer to use the actual name:
```php
	public function rules () {
		return [
			'creating.foo' => 'required'
		];
	}
```
```html
	<input wire:model="creating.foo" name="password" type="password">
```

**Context**
- Livewire version: 2.4.0
- Laravel version: 8.31.0
- Alpine version: 2.8.2
- Browser: Chrome, Safari, Firefox, Opera

**Comment by joshhanley** [comment]
@Coyote6 Livewire need to be able to serialise the data to json for the inputs to work. But as the properties are hidden, it can't display the values on the front end (but it can update them).

You would need to use something like `->makeVisible` to make that property available to the front end (I think you would have to put that in a dehydration hook or something).

But personally, I would not recommend binding to a password property on a model. Instead I would just create a dedicated password property, and add the value to the model on save.

Hope this helps!

**Comment by calebporzio** [comment]
Right on @joshhanley 

I think this is expected behavior, right?

**Comment by Coyote6** [comment]
Thanks guys.  I will just rename the property then unset it before saving.  Just took me a bit to figure out what was going on as everything else was working.   If you haven't already done so, you might add some documentation about this.  I easily could have missed it if you already have, as I am really bad about only reading the parts of the documentation I need at the moment and not the entire thing.  Appreciate the help!

**Reply by tonybyng** [reply]
Ive just hit this too so glad to find this response. In my case there are Accounting system setup codes that need to be set in the back end admin area but we've set them as hidden so when the api delivers the model to the front end they don't go over. Took me a while to realise what was happening but I guess if its serializing json data then it won't get the hidden stuff. Makes sense, just completely unexpected !
